File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/socialmedia/ReactionDao.java
package com.github.dedis.popstellar.repository.database.socialmedia;

import androidx.room.*;

import com.github.dedis.popstellar.model.objects.Reaction;
import com.github.dedis.popstellar.model.objects.security.MessageID;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

@Dao
public interface ReactionDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  Completable insert(ReactionEntity reactionEntity);

  @Query("SELECT reaction FROM reactions WHERE chirp_id = :chirpId")
  Single<List<Reaction>> getReactionsByChirpId(MessageID chirpId);
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/socialmedia/ReactionDao.kt
package com.github.dedis.popstellar.repository.database.socialmedia

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.github.dedis.popstellar.model.objects.Reaction
import com.github.dedis.popstellar.model.objects.security.MessageID
import io.reactivex.Completable
import io.reactivex.Single

@Dao
interface ReactionDao {
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  fun insert(reactionEntity: ReactionEntity): Completable

  @Query("SELECT reaction FROM reactions WHERE chirp_id = :chirpId")
  fun getReactionsByChirpId(chirpId: MessageID): Single<List<Reaction>?>
}
